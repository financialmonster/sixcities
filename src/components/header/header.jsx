import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';

const Header = ({isLoggedIn, email, isMain}) => {
    const spanClassName = (isLoggedIn) ? `header__user-name user__name` : `header__login`;
    let link;

    if(isMain) {
        link = (
            <a className="header__logo-link header__logo-link--active">
                <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41" />
            </a>
        );
    } else {
        link = (
            <Link to="/" className="header__logo-link">
                <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81"height="41" />
            </Link>
        );
    }

    return (
        <header className="header">
            <div className="container">
                <div className="header__wrapper">
                    <div className="header__left">
                        {link}
                    </div>
                    <nav className="header__nav">
                        <ul className="header__nav-list">
                            <li className="header__nav-item user">
                                <Link to={(isLoggedIn) ? `/favorites` : `/login`}
                                    className="header__nav-link header__nav-link--profile"
                                >
                                    <div className="header__avatar-wrapper user__avatar-wrapper"></div>
                                    <span className={spanClassName}>
                                        {(isLoggedIn) ? email : `Sign in`}
                                    </span>
                                </Link>
                            </li>
                        </ul>
                    </nav>
                </div>
            </div>
        </header>
    );
}

Header.propTypes = {
    isLoggedIn: PropTypes.bool.isRequired,
    email: PropTypes.string,
    isMain: PropTypes.bool.isRequired
}

const mapStateToProps = ({isLoggedIn, email}) => {
    return {
        isLoggedIn,
        email
    }
}

export {Header};
export default connect(mapStateToProps, () => ({}))(Header);
