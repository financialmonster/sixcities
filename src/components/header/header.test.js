import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import {Header} from './header.jsx';

describe(`Correctly renders the Header component`, () => {
    it(`when isLoggedIn and isMain properties are equal true, isAuthStatusLoading and isAuthStatusError properties are
        equal false`,
    () => {
        const tree = renderer.create(
            <Router>
                <Header isLoggedIn={true} isMain={true} isAuthStatusError={false} isAuthStatusLoading={false}
                    email='example@.ru'
                />
            </Router>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it(`when isLoggedIn property is equal true, isAuthStatusLoading, isMain and isAuthStatusError properties are equal
        false`,
    () => {
        const tree = renderer.create(
            <Router>
                <Header isLoggedIn={true} isMain={false} isAuthStatusError={false} isAuthStatusLoading={false}
                    email='example@.ru'
                />
            </Router>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it(`when isAuthStatusLoading and isMain properties are equal true, isAuthStatusError and isLoggedIn properties are
        equal false`,
    () => {
        const tree = renderer.create(
            <Router>
                <Header isLoggedIn={false} isMain={true} isAuthStatusError={false} isAuthStatusLoading={true}
                    email='example@.ru'
                />
            </Router>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it(`when isAuthStatusLoading property is equal true, isAuthStatusError, isLoggedIn, and isMain properties are equal
        false`,
    () => {
        const tree = renderer.create(
            <Router>
                <Header isLoggedIn={false} isMain={false} isAuthStatusError={false} isAuthStatusLoading={true}
                    email='example@.ru'
                />
            </Router>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it(`when isAuthStatusError and isMain properties are equal true, isAuthStatusLoading and isLoggedIn properties are
        equal false`,
    () => {
        const tree = renderer.create(
            <Router>
                <Header isLoggedIn={false} isMain={true} isAuthStatusError={true} isAuthStatusLoading={false}
                    email='example@.ru'
                />
            </Router>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it(`when isAuthStatusError property is equal true, isAuthStatusLoading, isMain and isLoggedIn properties are equal
        false`,
    () => {
        const tree = renderer.create(
            <Router>
                <Header isLoggedIn={false} isMain={false} isAuthStatusError={true} isAuthStatusLoading={false}
                    email='example@.ru'
                />
            </Router>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
