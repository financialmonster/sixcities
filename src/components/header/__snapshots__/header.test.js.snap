// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders the Header component when isAuthStatusError and isMain properties are equal true, isAuthStatusLoading and isLoggedIn properties are
        equal false 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <div>
              Authorization failed. Please, reload the page and try again
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Correctly renders the Header component when isAuthStatusError property is equal true, isAuthStatusLoading, isMain and isLoggedIn properties are equal
        false 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
          href="/"
          onClick={[Function]}
          title="To the main page"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <div>
              Authorization failed. Please, reload the page and try again
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Correctly renders the Header component when isAuthStatusLoading and isMain properties are equal true, isAuthStatusError and isLoggedIn properties are
        equal false 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <div>
              Authorization...
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Correctly renders the Header component when isAuthStatusLoading property is equal true, isAuthStatusError, isLoggedIn, and isMain properties are equal
        false 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
          href="/"
          onClick={[Function]}
          title="To the main page"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <div>
              Authorization...
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Correctly renders the Header component when isLoggedIn and isMain properties are equal true, isAuthStatusLoading and isAuthStatusError properties are
        equal false 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/favorites"
              onClick={[Function]}
              title="To the favorites page"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <span
                className="header__user-name user__name"
              >
                example@.ru
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Correctly renders the Header component when isLoggedIn property is equal true, isAuthStatusLoading, isMain and isAuthStatusError properties are equal
        false 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
          href="/"
          onClick={[Function]}
          title="To the main page"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/favorites"
              onClick={[Function]}
              title="To the favorites page"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <span
                className="header__user-name user__name"
              >
                example@.ru
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
